discovery-infra/test_infra/tools/static_network.py

```diff
@@ -71,6 +71,22 @@ def generate_static_network_data_from_tf(tf_folder: str) -> List[Dict[str, List[
     return network_config
 
 
+def get_name_to_mac_addresses_mapping(tf_folder: str) -> Dict[str, Set[str]]:
+    tfstate_json_file = os.path.join(tf_folder, "terraform.tfstate")
+    with open(tfstate_json_file) as _file:
+        tfstate = json.load(_file)
+    ret = dict()
+    for resource in tfstate["resources"]:
+        if resource["mode"] == "managed" and resource["type"] == "libvirt_domain":
+            for domain in resource["instances"]:
+                attributes = domain["attributes"]
+                macs = set()
+                for intf in attributes["network_interface"]:
+                    macs.add(intf["mac"].lower())
```

Comment:
I think that instead of returning a lower version here, you should validate in the calling function that both are lower case
```
            for mac in map(str.lower, host.macs()):
                for name, macs in mappings.items():
                    if mac in map(str.lower, macs):
                        return name
```

Reply:
[](https://github.com/openshift/assisted-test-infra/blob/223158e520161517ac2c68b88db486a357c014da/discovery-infra/test_infra/helper_classes/cluster_host.py#L31)